components/d2l-activity-editor/d2l-activity-quiz-editor/d2l-activity-quiz-manage-attempts-container.js

```diff
@@ -58,25 +58,17 @@ class ActivityQuizManageAttemptsContainer extends ActivityEditorWorkingCopyDialo
 	}
 
 	async _checkinDialog(e) {
-		await this.checkinDialog(e);
-
-		if (!this.opened) {
-			// Dialog successfully checked in
-			const checkedOutQuizEntity = this.checkedOutHref && store.get(this.checkedOutHref);
-			if (!checkedOutQuizEntity) return;
-			const { attemptsHref: checkedOutAttemptsHref } = checkedOutQuizEntity;
-
-			const dialogQuizEntity = this.dialogHref && store.get(this.dialogHref);
-			if (!dialogQuizEntity) return;
-			const { attemptsHref: dialogAttemptsHref } = dialogQuizEntity;
+		const { attemptsHref: dialogAttemptsHref } = (this.dialogHref && store.get(this.dialogHref)) || {};
+		const dialogAttemptsEntity = attemptsStore.get(dialogAttemptsHref);
+		// Wait for attempts entity PATCH requests to complete before checking in
+		if (dialogAttemptsEntity && dialogAttemptsEntity.saving) {
+			await dialogAttemptsEntity.saving;
+		}
```

Comment:
Should we be checking in if `!dialogAttemptsEntity`? Else I'd suggest
```suggestion
		if (!dialogAttemptsEntity) {
			return;
		}
		await dialogAttemptsEntity.saving;
```

Reply:
Agreed